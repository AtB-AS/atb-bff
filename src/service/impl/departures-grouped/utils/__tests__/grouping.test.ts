import mapQueryToGroups from '../grouping';

const favorites = [
  {
    destinationDisplay: {
      frontText: 'Trondheim S',
    },
    lineId: 'ATB:Line:200109141544126_24_200109141539610',
    quayId: 'NSR:Quay:74013',
  },
];

describe('service stops -> departure group utils', () => {
  // These aren't great tests ment for development.
  // Mostly to prevent accidental regressions / to enforce contracts
  // to easier track breaking changes.
  it('snapshot - should group data from venue with favorites', () => {
    const fixture = require('./fixture-venue-layer-favorites.json');
    expect(mapQueryToGroups(fixture, favorites)).toMatchSnapshot();
  });

  it('snapshot - should group data from venue with favorites with via data', () => {
    const fixture = require('./fixture-venue-layer-favorites-via.json');
    expect(mapQueryToGroups(fixture, favorites)).toMatchSnapshot();
  });
});
