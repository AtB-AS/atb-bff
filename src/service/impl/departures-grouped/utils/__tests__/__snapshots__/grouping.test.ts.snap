// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service stops -> departure group utils snapshot - should group data from venue with favorites 1`] = `
[
  {
    "quays": [
      {
        "group": [],
        "quay": {
          "__typename": "Quay",
          "description": null,
          "id": "NSR:Quay:74012",
          "latitude": 63.40766,
          "longitude": 10.403871,
          "name": "Sorgenfri",
          "publicCode": "",
          "situations": [],
          "stopPlace": {
            "__typename": "StopPlace",
            "description": "",
            "id": "NSR:StopPlace:43153",
            "latitude": 63.407925,
            "longitude": 10.403864,
            "name": "Sorgenfri",
          },
        },
      },
      {
        "group": [
          {
            "departures": [
              {
                "aimedTime": undefined,
                "bookingArrangements": undefined,
                "cancellation": undefined,
                "notices": [],
                "predictionInaccurate": false,
                "realtime": true,
                "serviceDate": undefined,
                "serviceJourneyId": "ATB:ServiceJourney:24_200109141599832_78_200109141539610",
                "situations": [],
                "time": undefined,
              },
              {
                "aimedTime": undefined,
                "bookingArrangements": undefined,
                "cancellation": undefined,
                "notices": [],
                "predictionInaccurate": false,
                "realtime": true,
                "serviceDate": undefined,
                "serviceJourneyId": "ATB:ServiceJourney:24_200109141599833_80_200109141539610",
                "situations": [],
                "time": undefined,
              },
              {
                "aimedTime": undefined,
                "bookingArrangements": undefined,
                "cancellation": undefined,
                "notices": [],
                "predictionInaccurate": false,
                "realtime": true,
                "serviceDate": undefined,
                "serviceJourneyId": "ATB:ServiceJourney:24_200109141599834_82_200109141539610",
                "situations": [],
                "time": undefined,
              },
              {
                "aimedTime": undefined,
                "bookingArrangements": undefined,
                "cancellation": undefined,
                "notices": [],
                "predictionInaccurate": false,
                "realtime": true,
                "serviceDate": undefined,
                "serviceJourneyId": "ATB:ServiceJourney:24_200109141599835_84_200109141539610",
                "situations": [],
                "time": undefined,
              },
              {
                "aimedTime": undefined,
                "bookingArrangements": undefined,
                "cancellation": undefined,
                "notices": [],
                "predictionInaccurate": false,
                "realtime": true,
                "serviceDate": undefined,
                "serviceJourneyId": "ATB:ServiceJourney:24_200109141599836_86_200109141539610",
                "situations": [],
                "time": undefined,
              },
            ],
            "lineInfo": {
              "destinationDisplay": {
                "__typename": "DestinationDisplay",
                "frontText": "Trondheim S",
              },
              "lineId": "ATB:Line:200109141544126_24_200109141539610",
              "lineName": "Trondheim S",
              "lineNumber": "24",
              "notices": [],
              "quayId": "NSR:Quay:74013",
              "transportMode": "bus",
              "transportSubmode": "localBus",
            },
          },
        ],
        "quay": {
          "__typename": "Quay",
          "description": null,
          "id": "NSR:Quay:74013",
          "latitude": 63.40819,
          "longitude": 10.403856,
          "name": "Sorgenfri",
          "publicCode": "",
          "situations": [],
          "stopPlace": {
            "__typename": "StopPlace",
            "description": "",
            "id": "NSR:StopPlace:43153",
            "latitude": 63.407925,
            "longitude": 10.403864,
            "name": "Sorgenfri",
          },
        },
      },
    ],
    "stopPlace": {
      "__typename": "StopPlace",
      "description": "",
      "id": "NSR:StopPlace:43153",
      "latitude": 63.407925,
      "longitude": 10.403864,
      "name": "Sorgenfri",
    },
  },
]
`;

exports[`service stops -> departure group utils snapshot - should group data from venue with favorites with via data 1`] = `
[
  {
    "quays": [
      {
        "group": [],
        "quay": {
          "__typename": "Quay",
          "description": null,
          "id": "NSR:Quay:74012",
          "latitude": 63.40766,
          "longitude": 10.403871,
          "name": "Sorgenfri",
          "publicCode": "",
          "situations": [],
          "stopPlace": {
            "__typename": "StopPlace",
            "description": "",
            "id": "NSR:StopPlace:43153",
            "latitude": 63.407925,
            "longitude": 10.403864,
            "name": "Sorgenfri",
          },
        },
      },
      {
        "group": [
          {
            "departures": [
              {
                "aimedTime": undefined,
                "bookingArrangements": undefined,
                "cancellation": undefined,
                "notices": [],
                "predictionInaccurate": false,
                "realtime": true,
                "serviceDate": undefined,
                "serviceJourneyId": "ATB:ServiceJourney:24_200109141599832_78_200109141539610",
                "situations": [],
                "time": undefined,
              },
              {
                "aimedTime": undefined,
                "bookingArrangements": undefined,
                "cancellation": undefined,
                "notices": [],
                "predictionInaccurate": false,
                "realtime": true,
                "serviceDate": undefined,
                "serviceJourneyId": "ATB:ServiceJourney:24_200109141599833_80_200109141539610",
                "situations": [],
                "time": undefined,
              },
              {
                "aimedTime": undefined,
                "bookingArrangements": undefined,
                "cancellation": undefined,
                "notices": [],
                "predictionInaccurate": false,
                "realtime": true,
                "serviceDate": undefined,
                "serviceJourneyId": "ATB:ServiceJourney:24_200109141599834_82_200109141539610",
                "situations": [],
                "time": undefined,
              },
              {
                "aimedTime": undefined,
                "bookingArrangements": undefined,
                "cancellation": undefined,
                "notices": [],
                "predictionInaccurate": false,
                "realtime": true,
                "serviceDate": undefined,
                "serviceJourneyId": "ATB:ServiceJourney:24_200109141599835_84_200109141539610",
                "situations": [],
                "time": undefined,
              },
              {
                "aimedTime": undefined,
                "bookingArrangements": undefined,
                "cancellation": undefined,
                "notices": [],
                "predictionInaccurate": false,
                "realtime": true,
                "serviceDate": undefined,
                "serviceJourneyId": "ATB:ServiceJourney:24_200109141599836_86_200109141539610",
                "situations": [],
                "time": undefined,
              },
            ],
            "lineInfo": {
              "destinationDisplay": {
                "__typename": "DestinationDisplay",
                "frontText": "Trondheim S",
              },
              "lineId": "ATB:Line:200109141544126_24_200109141539610",
              "lineName": "Trondheim S",
              "lineNumber": "24",
              "notices": [],
              "quayId": "NSR:Quay:74013",
              "transportMode": "bus",
              "transportSubmode": "localBus",
            },
          },
        ],
        "quay": {
          "__typename": "Quay",
          "description": null,
          "id": "NSR:Quay:74013",
          "latitude": 63.40819,
          "longitude": 10.403856,
          "name": "Sorgenfri",
          "publicCode": "",
          "situations": [],
          "stopPlace": {
            "__typename": "StopPlace",
            "description": "",
            "id": "NSR:StopPlace:43153",
            "latitude": 63.407925,
            "longitude": 10.403864,
            "name": "Sorgenfri",
          },
        },
      },
    ],
    "stopPlace": {
      "__typename": "StopPlace",
      "description": "",
      "id": "NSR:StopPlace:43153",
      "latitude": 63.407925,
      "longitude": 10.403864,
      "name": "Sorgenfri",
    },
  },
]
`;
